src/shared/services/katex.service.ts

```diff
@@ -0,0 +1,168 @@
+import { Injectable } from '@angular/core';
+import { List, Map } from 'immutable';
+import { render, ParseError } from 'katex';
+
+import { JSONSchema } from '../interfaces';
+
+@Injectable()
+export class KatexService {
+  private delimiters = [
+    { left: '$$', right: '$$', display: true },
+    { left: '$', right: '$', display: false },
+    // FIXME check if we need the ones below
```

Comment:
I am just not sure in all our records all the ways math formulas can appear. They can be in between $ or $$ and also I have seen in between `\begin{equation}` and `\end{equation}`